CircleCI tests sporadically fail

**Detailed Description**
Tests will randomly fail in CircleCI with `SurefireBooterForkException: The forked VM terminated without properly saying goodbye. VM crash or System.exit called?`. An example of this can be seen for master at https://circleci.com/gh/hashgraph/hedera-mirror-node/3134.

To address, we will need to modify CircleCI to store_artifacts the surefire logs (not just reports) so we can view them and investigate. Maybe we need to increase or decrease JVM memory to run in CircleCI.

**Actual Behavior**
Push code, watch tests fail

```console
[INFO] Results:
[INFO] 
[INFO] Tests run: 27, Failures: 0, Errors: 0, Skipped: 0
[INFO] 
[INFO] ------------------------------------------------------------------------
[INFO] Reactor Summary for Hedera Mirror Node 0.7.0-rc1:
[INFO] 
[INFO] Hedera Mirror Node ................................. SUCCESS [  0.003 s]
[INFO] Hedera Mirror Node Importer ........................ FAILURE [ 38.740 s]
[INFO] Hedera Mirror Node Coverage ........................ SKIPPED
[INFO] Hedera Mirror Data Generator ....................... SKIPPED
[INFO] Hedera Mirror Node Protobuf ........................ SKIPPED
[INFO] Hedera Mirror Node GRPC API ........................ SKIPPED
[INFO] Hedera Mirror Node REST API ........................ SKIPPED
[INFO] Hedera Mirror Node Test ............................ SKIPPED
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  39.211 s
[INFO] Finished at: 2020-03-24T18:28:13Z
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-surefire-plugin:2.22.2:test (default-test) on project hedera-mirror-importer: There are test failures.
[ERROR] 
[ERROR] Please refer to /root/project/hedera-mirror-importer/target/surefire-reports for the individual test results.
[ERROR] Please refer to dump files (if any exist) [date].dump, [date]-jvmRun[N].dump and [date].dumpstream.
[ERROR] The forked VM terminated without properly saying goodbye. VM crash or System.exit called?
[ERROR] Command was /bin/sh -c cd /root/project/hedera-mirror-importer && /opt/java/openjdk/bin/java -javaagent:/root/.m2/repository/org/jacoco/org.jacoco.agent/0.8.5/org.jacoco.agent-0.8.5-runtime.jar=destfile=/root/project/hedera-mirror-importer/target/jacoco.exec -Xms512m -jar /root/project/hedera-mirror-importer/target/surefire/surefirebooter13392143665330743611.jar /root/project/hedera-mirror-importer/target/surefire 2020-03-24T18-27-46_978-jvmRun1 surefire8872744923615715811tmp surefire_017727579870895475179tmp
[ERROR] Error occurred in starting fork, check output in log
[ERROR] Process Exit Code: 137
[ERROR] org.apache.maven.surefire.booter.SurefireBooterForkException: The forked VM terminated without properly saying goodbye. VM crash or System.exit called?
[ERROR] Command was /bin/sh -c cd /root/project/hedera-mirror-importer && /opt/java/openjdk/bin/java -javaagent:/root/.m2/repository/org/jacoco/org.jacoco.agent/0.8.5/org.jacoco.agent-0.8.5-runtime.jar=destfile=/root/project/hedera-mirror-importer/target/jacoco.exec -Xms512m -jar /root/project/hedera-mirror-importer/target/surefire/surefirebooter13392143665330743611.jar /root/project/hedera-mirror-importer/target/surefire 2020-03-24T18-27-46_978-jvmRun1 surefire8872744923615715811tmp surefire_017727579870895475179tmp
[ERROR] Error occurred in starting fork, check output in log
[ERROR] Process Exit Code: 137
[ERROR] 	at org.apache.maven.plugin.surefire.booterclient.ForkStarter.fork(ForkStarter.java:669)
[ERROR] 	at org.apache.maven.plugin.surefire.booterclient.ForkStarter.run(ForkStarter.java:282)
[ERROR] 	at org.apache.maven.plugin.surefire.booterclient.ForkStarter.run(ForkStarter.java:245)
[ERROR] 	at org.apache.maven.plugin.surefire.AbstractSurefireMojo.executeProvider(AbstractSurefireMojo.java:1183)
[ERROR] 	at org.apache.maven.plugin.surefire.AbstractSurefireMojo.executeAfterPreconditionsChecked(AbstractSurefireMojo.java:1011)
[ERROR] 	at org.apache.maven.plugin.surefire.AbstractSurefireMojo.execute(AbstractSurefireMojo.java:857)
[ERROR] 	at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo(DefaultBuildPluginManager.java:137)
[ERROR] 	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:210)
[ERROR] 	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:156)
[ERROR] 	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:148)
[ERROR] 	at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:117)
[ERROR] 	at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:81)
[ERROR] 	at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build(SingleThreadedBuilder.java:56)
[ERROR] 	at org.apache.maven.lifecycle.internal.LifecycleStarter.execute(LifecycleStarter.java:128)
[ERROR] 	at org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:305)
[ERROR] 	at org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:192)
[ERROR] 	at org.apache.maven.DefaultMaven.execute(DefaultMaven.java:105)
[ERROR] 	at org.apache.maven.cli.MavenCli.execute(MavenCli.java:957)
[ERROR] 	at org.apache.maven.cli.MavenCli.doMain(MavenCli.java:289)
[ERROR] 	at org.apache.maven.cli.MavenCli.main(MavenCli.java:193)
[ERROR] 	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
[ERROR] 	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
[ERROR] 	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
[ERROR] 	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
[ERROR] 	at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced(Launcher.java:282)
[ERROR] 	at org.codehaus.plexus.classworlds.launcher.Launcher.launch(Launcher.java:225)
[ERROR] 	at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode(Launcher.java:406)
[ERROR] 	at org.codehaus.plexus.classworlds.launcher.Launcher.main(Launcher.java:347)
[ERROR] 	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
[ERROR] 	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
[ERROR] 	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
[ERROR] 	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
[ERROR] 	at org.apache.maven.wrapper.BootstrapMainStarter.start(BootstrapMainStarter.java:39)
[ERROR] 	at org.apache.maven.wrapper.WrapperExecutor.execute(WrapperExecutor.java:122)
[ERROR] 	at org.apache.maven.wrapper.MavenWrapperMain.main(MavenWrapperMain.java:61)
```

**Expected Behavior**
Tests to succeed

**Environment:**
- Version: master

**Additional Context**

--- a/.circleci/config.yml
+++ b/.circleci/config.yml
@@ -93,6 +93,12 @@ jobs:
       - run:
           name: Running maven (validate, compile, test, package)
           command: ./mvnw ${MAVEN_CLI_OPTS} package
+      - store_artifacts:
+          path: hedera-mirror-importer/target/surefire-reports
+          destination: /importer-surefire
+      - store_artifacts:
+          path: hedera-mirror-grpc/target/surefire-reports
+          destination: /grpc-surefire
       - store_test_results:
           path: hedera-mirror-importer/target/surefire-reports
       - run:
@@ -123,12 +129,6 @@ jobs:
             mkdir -p ${WORKSPACE}/artifacts
             tar -czf ${WORKSPACE}/artifacts/${NAME}.tgz -C ${WORKSPACE} ${NAME}
       - *persist_artifacts
-      - run:
-          name: Storing surefire logs for debugging
-          command: |
-            mv hedera-mirror-importer/target/surefire-reports ${WORKSPACE}/importer-surefire
-      - store_artifacts:
-          path: ${WORKSPACE}/importer-surefire
 
   perf_maven:
     environment:

--- a/.circleci/config.yml
+++ b/.circleci/config.yml
@@ -124,9 +124,9 @@ jobs:
             tar -czf ${WORKSPACE}/artifacts/${NAME}.tgz -C ${WORKSPACE} ${NAME}
       - *persist_artifacts
       - run:
-          - Storing surefire logs for debugging
-          - command: |
-              mv hedera-mirror-importer/target/surefire-reports ${WORKSPACE}/importer-surefire
+          name: Storing surefire logs for debugging
+          command: |
+            mv hedera-mirror-importer/target/surefire-reports ${WORKSPACE}/importer-surefire
       - store_artifacts:
           path: ${WORKSPACE}/importer-surefire
 

--- a/.circleci/config.yml
+++ b/.circleci/config.yml
@@ -92,7 +92,7 @@ jobs:
       - *save_maven_cache
       - run:
           name: Running maven (validate, compile, test, package)
-          command: ./mvnw ${MAVEN_CLI_OPTS} package -Xmx1024m
+          command: ./mvnw ${MAVEN_CLI_OPTS} package
       - store_artifacts:
           path: hedera-mirror-importer/target/surefire-reports
           destination: /importer-surefire

--- a/.circleci/config.yml
+++ b/.circleci/config.yml
@@ -123,6 +123,12 @@ jobs:
             mkdir -p ${WORKSPACE}/artifacts
             tar -czf ${WORKSPACE}/artifacts/${NAME}.tgz -C ${WORKSPACE} ${NAME}
       - *persist_artifacts
+      - run:
+          - Storing surefire logs for debugging
+          - command: |
+              mv hedera-mirror-importer/target/surefire-reports ${WORKSPACE}/importer-surefire
+      - store_artifacts:
+          path: ${WORKSPACE}/importer-surefire
 
   perf_maven:
     environment:

--- a/.circleci/config.yml
+++ b/.circleci/config.yml
@@ -92,7 +92,7 @@ jobs:
       - *save_maven_cache
       - run:
           name: Running maven (validate, compile, test, package)
-          command: ./mvnw ${MAVEN_CLI_OPTS} package
+          command: ./mvnw ${MAVEN_CLI_OPTS} package -Xmx1024m
       - store_artifacts:
           path: hedera-mirror-importer/target/surefire-reports
           destination: /importer-surefire
@@ -177,6 +177,9 @@ jobs:
           environment:
             TEST_DB_HOST: "127.0.0.1"
             TEST_DB_NAME: "mirror_node_integration"
+      - store_artifacts:
+          path: hedera-mirror-rest/target/jest-junit
+          destination: /rest-jest
       - store_test_results:
           path: hedera-mirror-rest/target/jest-junit
       - run:

--- a/pom.xml
+++ b/pom.xml
@@ -77,11 +77,12 @@
                 <plugin>
                     <groupId>org.apache.maven.plugins</groupId>
                     <artifactId>maven-surefire-plugin</artifactId>
+                    <version>3.0.0-M4</version>
                     <configuration>
                         <!-- Jacoco prepare-agent builds some command-line params without -->
                         <!-- which jacoco will not instrument. Hence it is important to add -->
                         <!-- those command-line params here (${argLine} holds those params) -->
-                        <argLine>${argLine} -Xms512m</argLine>
+                        <argLine>${argLine} -Xms512m -Xmx1024m</argLine>
                     </configuration>
                 </plugin>
                 <plugin>

